### PR TITLE
Exercises from 007 to 011

### PR DESCRIPTION
**Pull request by Dreamless**
Exercises by "While Loops, Switch statements, Objects, Null and Undefined, Type Conversion, Template Strings and Ternary Operator"

### CODE REVIEW

File: javascript/007-high-score-board/high-score-board.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Increases a player's score by the given amount.
+ *
+ * @param {Record<string, number>} scoreBoard
+ * @param {string} player
+ * @param {number} points
+ * @returns {Record<string, number>} updated score board
+ */
+export function updateScore(scoreBoard, player, points) {
+  scoreBoard[player] = scoreBoard[player] + points;
```
Comment (fantgeass) [inline javascript/007-high-score-board/high-score-board.js:47]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Addition_assignment

File: javascript/007-high-score-board/high-score-board.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * Applies 100 bonus points to all players on the board.
+ *
+ * @param {Record<string, number>} scoreBoard
+ * @returns {Record<string, number>} updated score board
+ */
+export function applyMondayBonus(scoreBoard) {
+  const mondayBonus = 100;
+
+  for (let player in scoreBoard) {
+    if (Object.prototype.hasOwnProperty.call(scoreBoard, player)) {
```
Comment (fantgeass) [inline javascript/007-high-score-board/high-score-board.js:61]: Explain please why you need this condition

File: javascript/008-mixed-juices/mixed-juices.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  let wadgesCount = 0;
+  let limesCount = 0;
+
+  const limeWedges = {
+    small: 6,
+    medium: 8,
+    large: 10
+  };
+
+  while (wadgesCount <= wedgesNeeded) {
```
Comment (frozenspider) [inline javascript/008-mixed-juices/mixed-juices.js:52]: Think about the case when `wedgesCount == wedgesNeeded`

File: javascript/008-mixed-juices/mixed-juices.spec.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      'large',
+      'medium',
+      'small',
+      'large',
+      'large',
+      'medium',
+    ];
+    expect(limesToCut(42, limes)).toBe(6);
+
+    expect(limesToCut(4, ['medium', 'small'])).toBe(1);
```
Comment (frozenspider) [inline javascript/008-mixed-juices/mixed-juices.spec.js:37]: Add a test when number of lime wedges we need to cut is exactly satisfied by taking a few limes from an array.

File: javascript/008-mixed-juices/mixed-juices.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+export function remainingOrders(timeLeft, orders) {
+  let i = 0;
+  let firstShiftTime = 0;
+  const remaningOrders = [];
+
+  while (i < orders.length) {
+    if (firstShiftTime < timeLeft) {
+      firstShiftTime += timeToMixJuice(orders[i]);
+    } else {
+      remaningOrders.push(orders[i]);
```
Comment (frozenspider) [inline javascript/008-mixed-juices/mixed-juices.js:81]: You can make code way cleaner that this. All we need is to find index of the first cocktail we shouldn't be making, after than we take all of them.
You don't need `remaningOrders` accumulator for that.

File: javascript/009-amusement-park/amusement-park.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+ * @param {Record<string, string|null>} tickets
+ * @param {string} ticketId
+ * @returns {string} ticket status
+ */
+export function ticketStatus(tickets, ticketId) {
+  if (tickets[ticketId] === undefined) {
+    return 'unknown ticket id';
+  }
+
+  return tickets[ticketId] !== null ? `sold to ${tickets[ticketId]}` : 'not sold';
```
Comment (frozenspider) [inline javascript/009-amusement-park/amusement-park.js:39]: Don't mix if-statements and if-expressions together, that looks weird and is hard to read.

File: javascript/011-custom-signs/custom-signs.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+ * @param {string} sign
+ * @param {string} currency
+ *
+ * @returns {string} cost to create the sign
+ */
+
+export function costOf(sign, currency) {
+  const baseSignCost = 20;
+  const letterCost = 2;
+  const calculatedPrice = baseSignCost + (sign.trim().length * letterCost);
```
Comment (frozenspider) [inline javascript/011-custom-signs/custom-signs.js:61]: Re-read the README about how whitespaces should be handled. Add a test for that.

**Review by frozenspider — CHANGES_REQUESTED** [review]

File: javascript/007-high-score-board/high-score-board.js (hunk 2 above)
Comment (Dreamless) [reply javascript/007-high-score-board/high-score-board.js:61]: It's EsLint rule look below. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
https://eslint.org/docs/latest/rules/no-prototype-builtins

File: javascript/008-mixed-juices/mixed-juices.spec.js (hunk 4 above)
Comment (frozenspider) [reply javascript/008-mixed-juices/mixed-juices.spec.js:37]: This still stands

**Review by frozenspider — CHANGES_REQUESTED** [review]

File: javascript/008-mixed-juices/mixed-juices.js [hunk 8]
```diff
@@ -70,19 +65,16 @@ export function limesToCut(wedgesNeeded, limes) {
...
-      remaningOrders.push(orders[i]);
-    }
+  if (orders.length === 0) {
+    return [];
+  }
 
-    i++;
+  let firstShiftTime = 0;
+  let orderIndex = 0;
+  while (firstShiftTime < timeLeft) {
```
[the remark this answers is not in the document]
Comment (frozenspider) [reply javascript/008-mixed-juices/mixed-juices.js:74]: This comment still stands

**Review by frozenspider — APPROVED** [review]

**Pull request merged**